tempodb/backend/backend.go

```diff
@@ -39,8 +39,9 @@ type Reader interface {
 	Read(ctx context.Context, name string, blockID uuid.UUID, tenantID string, shouldCache bool) ([]byte, error)
 	// StreamReader is for streaming entire objects from the backend.  It is expected this will _not_ be cached.
 	StreamReader(ctx context.Context, name string, blockID uuid.UUID, tenantID string) (io.ReadCloser, int64, error)
-	// ReadRange is for reading parts of large objects from the backend.  It is expected this will _not_ be cached.
-	ReadRange(ctx context.Context, name string, blockID uuid.UUID, tenantID string, offset uint64, buffer []byte) error
+	// ReadRange is for reading parts of large objects from the backend.
+	// There will be an attempt to retrieve this from cache if shouldCache is true. Cache key will be tenantID:blockID:offset:bufferLength
```

Comment:
> There will be an attempt to retrieve this from cache if shouldCache is true

I think this could come up as unexpected behaviour from reading the param name. I'd expect to read always try to read from the cache, but only cache in certain scenarios.

I feel like using a wrapper for the cases in which using a cache is possible is a bit clearer (e.g. `NewCachedReader(r Reader)`

Reply:
Yes, there is a design choice in place in here about whether we want the application to handle storing data in cache (similar to the approach you have mentioned `NewCachedReader`) vs. passing a `shouldCache` parameter and having the underlying reader worry about how to handle caching those objects. 

We have opted for the latter - where Tempo can just specify which objects should be cached, and the underlying backend reader will handle that. If there are strong opinions against this approach we can discuss how the alternate could work. Note that we are already using the `shouldCache` design with `Read(..)` so that will have to be changed as well - https://github.com/grafana/tempo/blob/29ee11386fcd49fa3deb6cf8772d45f22a5fdf91/tempodb/backend/backend.go#L38-L39

Reply:
If it's been discussed and in use already, I'm fine with it. I only find the name of the param and its behaviour slightly confusing, but it's not a big deal IMO.

Thanks for clarifying.

Reply:
If the naming is confusing we could change it, open to suggestions :)